pkg/trace/call.go

```diff
@@ -107,6 +106,13 @@ func SetCallError(ctx context.Context, err error) error {
 // rethrows them.  Any panics are converted to errors and cause error
 // logging to happen (as do any SetCallStatus calls)
 func EndCall(ctx context.Context) {
+	callInfo := callTracker.Info(ctx)
+	if callInfo == nil {
+		// There was no call associated with this context, so we can't
+		// do anything.
```

Comment:
Is it correct to skip the `defer End(ctx)` in this case? I assume it's a no-op without a call?

Reply:
Yes, while it would noop it doesn't make sense to hit that code path if we have no trace information -- that's equivalent to having no trace at all, so one can't end it.